carbon-design-system/carbon-addons-iot-react
# fix(HierarchyList): scroll only if needed and support secondary nodes

**Summary**

- Scrolling was always happening on item selection regardless of whether the item was already in view or not causing some weird jumping
- `secondaryValue` was not accepting nodes in the list

**Change List (commits, features, bugs, etc)**

- Added the `scroll-into-view-if-needed` package to check if the node is visible or not, before attempting to scroll to it
- Allowed `secondaryValue` to accept nodes
- Added string type check for searching secondary values
- Updated 1 HierarchyList story to use a custom component as `secondaryValue`


**Acceptance Test (how to verify the PR)**

- Go to /?path=/story/watson-iot-hierarchylist--with-default-selected-id in chrome, firefox, and safari and ensure the item is scrolled to on initial render, then continue to select other items and ensure no scrolling occurs
- Go to /?path=/story/watson-iot-hierarchylist--with-mixed-hierarchies, open the 'Tasks' category, and see that the InlineLoading component is successfully rendering, then use the search bar and ensure searching still works (this ensures I didn't break the search by adding the node)
